### PR TITLE
CVE-2014-1746 and CVE-2016-1616

### PR DESCRIPTION
**Pull request by bentgarbacz**

### CODE REVIEW

File: cves/CVE-2014-1746.yml [hunk 1]
```diff
@@ -95,20 +102,20 @@ discovered:
...
 subsystem:
   question: |
     What subsystems was the mistake in?
 
     Look at the path of the source code files code that were fixed to get
     directory names. Look at comments in the code. Look at the bug reports how
     the bug report was tagged. Examples: "clipboard", "gpu", "ssl", "speech", "renderer"
-  answer: 
-  name: 
+  answer: Based on the description in the CVE.
```
Comment (mesternefeld) [inline cves/CVE-2014-1746.yml]: Give more details about media filters and a look into what it is.

File: cves/CVE-2016-1616.yml [hunk 2]
```diff
@@ -32,7 +32,9 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: 
+description: |
+  There was a vulnerability where an attacker could spoof a URL using an unfocused
```
Comment (mesternefeld) [inline cves/CVE-2016-1616.yml]: This is a good start! Add some more details. This sounds interesting and I'd love to hear more about it.

File: cves/CVE-2016-1616.yml [hunk 3]
```diff
@@ -69,7 +71,7 @@ upvotes:
 unit_tested:
   question: |
     Were automated unit tests involved in this vulnerability?
-    Was the original code unit tested, or not unit tested? Did the fix involve
+    Was the original code unit tested, or nooverflowt unit tested? Did the fix involve
```
Comment (mesternefeld) [inline cves/CVE-2016-1616.yml]: I think this shouldn't be here. Please revert

File: cves/CVE-2016-1616.yml [hunk 4]
```diff
@@ -96,20 +98,20 @@ discovered:
...
 subsystem:
   question: |
     What subsystems was the mistake in?
 
     Look at the path of the source code files code that were fixed to get
     directory names. Look at comments in the code. Look at the bug reports how
     the bug report was tagged. Examples: "clipboard", "gpu", "ssl", "speech", "renderer"
-  answer: 
-  name: 
+  answer: Based on the description in the CVE and the source code.
```
Comment (mesternefeld) [inline cves/CVE-2016-1616.yml]: Same like what I said above. And is the subsystem a custom button? I wouldn't expect something to be in a custom button directory.

File: cves/CVE-2014-1746.yml [hunk 5]
```diff
@@ -95,20 +102,20 @@ discovered:
 
     If there is no evidence as to how this vulnerability was found, then you may
     leave the entries blank except for "answer". Write down where you looked in "answer".
-  answer: 
-  date: 
-  automated: 
-  google: 
-  contest: 
+  answer: Someone ran into an overflow error and reported it.
```
Comment (MattThyng) [inline cves/CVE-2014-1746.yml]: Do you know who reported it? Was it a Google employee?

File: cves/CVE-2014-1746.yml [hunk 6]
```diff
@@ -198,4 +211,9 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer: 
+  answer: |
+    This bug was caused by developers not realizing that changing the data type to
+    int would not allow for the variable to be sufficiently large. When making any
+    changes, developers should take time to contemplate why things were the way they
+    were. Unless the developer in intimately knowledgeable about the source code,
+    it's a good idea to consult colleagues about potential changes.
```
Comment (MattThyng) [inline cves/CVE-2014-1746.yml]: May want to mention that this is a coding mistake

File: cves/CVE-2016-1616.yml [hunk 7]
```diff
@@ -32,7 +32,9 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: 
+description: |
+  There was a vulnerability where an attacker could spoof a URL using an unfocused
+  button in the UI.
```
Comment (MattThyng) [inline cves/CVE-2016-1616.yml]: You may want to explain what a unfocused button is

File: cves/CVE-2016-1616.yml [hunk 8]
```diff
@@ -56,8 +58,8 @@ fixes:
 - :commit: 297ae873b471a46929ea39697b121c0b411434ee
   :note: ''
 vccs:
-- :commit:
-  :note: 
+- :commit: 2c36e1fa592c341f27f758cf8b6770957c9bfdd4
+  :note: First commit with custom_button.cc, where the bug was located
```
Comment (MattThyng) [inline cves/CVE-2016-1616.yml]: Do you know what the purpose of this commit was?

File: cves/CVE-2016-1616.yml [hunk 9]
```diff
@@ -96,20 +98,20 @@ discovered:
 
     If there is no evidence as to how this vulnerability was found, then you may
     leave the entries blank except for "answer". Write down where you looked in "answer".
-  answer: 
-  date: 
-  automated: 
-  google: 
-  contest: 
+  answer: A member of the community found the bug and reported it.
```
Comment (MattThyng) [inline cves/CVE-2016-1616.yml]: You may want to mention who reported it

File: cves/CVE-2016-1616.yml [hunk 10]
```diff
@@ -199,4 +212,9 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer: 
+  answer: |
+    The lesson to be taken from this is that determining obscure edge cases can
+    help determine potentially dangerous vulnerabilities. Oversights are inevitable,
+    so bounty programs a good way to get as many eyes on your code as possible. If
+    this vulnerability wasn't reported by a community member, it could have languished
+    in the source code for years to come.
```
Comment (MattThyng) [inline cves/CVE-2016-1616.yml]: Was this a mistake in the planning, or a coding mistake? Miscommunication?

File: cves/CVE-2014-1746.yml (hunk 6 above)
Comment (johnbyrneRIT) [inline cves/CVE-2014-1746.yml]: You mention the switch to AVIO prompting changes. Was the mistake here a miscommunication of the new requirements where the developer wasn't aware, or just was it just a coding mistake?

File: cves/CVE-2014-1746.yml [hunk 11]
```diff
@@ -132,10 +141,12 @@ major_events:
 
     The event doesn't need to be directly related to this vulnerability, rather,
     we want to capture what the development team was dealing with at the time.
-  answer: 
+  answer: |
+    The dev team switched to AVIO from a custom FFmpeg URLProtocol handler. This necessitated
+    many changes, one of which caused the bug.
```
Comment (johnbyrneRIT) [inline cves/CVE-2014-1746.yml]: Is it clear what change specifically caused the bug and why?

File: cves/CVE-2016-1616.yml [hunk 12]
```diff
@@ -121,8 +123,11 @@ interesting_commits:
     If there are no interesting commits, demonstrate that you completed this section by explaining what happened between the VCCs and the fix.
   answer:
   commits:
-  - commit: 
-    note: 
+  - commit: f4856f1766103bcdf8b2625a4cf606b710bae9c2
+    note: |
+      In this commit, they changed how accelerators are handled. The bug dealt with how
+      custom buttons interact with accelerators, so they could have caught the bug years
+      earlier.
```
Comment (johnbyrneRIT) [inline cves/CVE-2016-1616.yml]: Was the change in accelerators part of what caused the vulnerability, or part of how they fixed it?

File: cves/CVE-2016-1616.yml [hunk 13]
```diff
@@ -185,8 +195,11 @@ lessons:
     applies: 
     note: 
   complex_inputs:
-    applies: 
-    note: 
+    applies: true
+    note: |
+      This vulnerability concerns an edge case that was not discovered until years
+      after release, so it's safe to say that the input required from a user to exploit
+      this vulnerability is complex.
```
Comment (johnbyrneRIT) [inline cves/CVE-2016-1616.yml]: This makes me curious as to what the inputs that could cause exploit this vulnerability were, and why unfocused buttons specifically caused the error.

File: cves/CVE-2014-1746.yml (hunk 1 above)
Comment (bentgarbacz) [reply cves/CVE-2014-1746.yml]: added more details

File: cves/CVE-2016-1616.yml (hunk 2 above)
Comment (bentgarbacz) [reply cves/CVE-2016-1616.yml]: Added some more details

File: cves/CVE-2016-1616.yml (hunk 3 above)
Comment (bentgarbacz) [reply cves/CVE-2016-1616.yml]: No idea how that happened, fixed

File: cves/CVE-2016-1616.yml (hunk 4 above)
Comment (bentgarbacz) [reply cves/CVE-2016-1616.yml]: The directory is called "button", but I said the subsystem was "custom button" because that was the only type of button affected by this vulnerability. Given that, do you think I should change the subsystem? Also, added more details.

File: cves/CVE-2014-1746.yml (hunk 5 above)
Comment (bentgarbacz) [reply cves/CVE-2014-1746.yml]: Added the name of the reporter

File: cves/CVE-2014-1746.yml (hunk 6 above)
Comment (bentgarbacz) [reply cves/CVE-2014-1746.yml]: added

File: cves/CVE-2016-1616.yml (hunk 7 above)
Comment (bentgarbacz) [reply cves/CVE-2016-1616.yml]: I added an explanation

File: cves/CVE-2016-1616.yml (hunk 8 above)
Comment (bentgarbacz) [reply cves/CVE-2016-1616.yml]: This commit added the file to it's current directory, so it's the vcc by default. I added some more details that hopefully clear it up.

File: cves/CVE-2016-1616.yml (hunk 9 above)
Comment (bentgarbacz) [reply cves/CVE-2016-1616.yml]: Added a name

File: cves/CVE-2016-1616.yml (hunk 10 above)
Comment (bentgarbacz) [reply cves/CVE-2016-1616.yml]: I would say a planning mistake, added rational in the cve.

File: cves/CVE-2014-1746.yml [hunk 14]
```diff
@@ -32,7 +32,12 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: 
+description: |
+  A function relies on an integer data type that is not large enough. This
```
[the remark this answers is not in the document]
Comment (bentgarbacz) [reply cves/CVE-2014-1746.yml]: I don't think a regular user would ever encounter this bug as it would require a vector with bounds larger than the size of an int, which seems extremely unlikely for everyday use.

File: cves/CVE-2016-1616.yml (hunk 12 above)
Comment (bentgarbacz) [reply cves/CVE-2016-1616.yml]: Neither, I thought this commit was interesting because they were clearly working in a section of code that had a vulnerability but glossed over it somehow.

File: cves/CVE-2014-1746.yml (hunk 11 above)
Comment (bentgarbacz) [reply cves/CVE-2014-1746.yml]: There was a single change that caused this bug, which was changing a data type to int.

File: cves/CVE-2014-1746.yml (hunk 6 above)
Comment (bentgarbacz) [reply cves/CVE-2014-1746.yml]: The issue seems to be a miscommunication through a lack of documenting requirements. If the original developers had provided rational as to why a certain data type was chosen, then the issue probably wouldn't have existed after the switch to AVIO.

File: cves/CVE-2016-1616.yml (hunk 13 above)
Comment (bentgarbacz) [reply cves/CVE-2016-1616.yml]: Input wasn't checked for unfocused buttons like it was for focused buttons, so malicious users could exploit this if they were able to actuate an unfocused button through the use of an accelerator.

File: cves/CVE-2014-1746.yml [hunk 15]
```diff
@@ -77,9 +84,9 @@ unit_tested:
 
     For the "fix" answer below, check if the fix for the vulnerability involves
     adding or improving an automated test to ensure this doesn't happen again.
-  answer: 
-  code: 
-  fix: 
+  answer: This fix was tested and a test was created.
```
Comment (blackpan2) [inline cves/CVE-2014-1746.yml]: With you saying that "code: true" on the next line, means there was tests in the repo that were around this section of code. Could you explain how these tests were insufficient (such as testing the relevant edge case), or if there were no tests update line 88 to be false.

File: cves/CVE-2016-1616.yml [hunk 16]
```diff
@@ -96,20 +104,23 @@ discovered:
...
 
     Look at the path of the source code files code that were fixed to get
     directory names. Look at comments in the code. Look at the bug reports how
     the bug report was tagged. Examples: "clipboard", "gpu", "ssl", "speech", "renderer"
-  answer: 
-  name: 
+  answer: |
+    Based on the description in the CVE and the source code. The issue is that when
+    unfocused buttons are allowed to interact with accelerators, or keyboard shortcuts,
+    a malicious actor could force a user to navigate to an arbitrary website.
```
Comment (blackpan2) [inline cves/CVE-2016-1616.yml]: "a malicious actor could force a user to navigate to an arbitrary website" This is better suited as a final explanation in your description making it clear the threat this vulnerability represents. It is a little out of place here where you are answering the question of "How did you come up with the subsystem you indicate on the line below?".
